### PR TITLE
feature/test-cases/api-integration

### PR DESCRIPTION
**Pull request by AsmitaMargaje**
clean the code and integrate the api :

1. test darft
2. review test draft
3. test case review
4. test bank

### CODE REVIEW

File: src/redux/slices/testCases/downloadFormatFile/index.js [hunk 1]
```diff
@@ -51,7 +76,7 @@ const downloadFormatSlice = createSlice({
       .addCase(getProjectModuleMasterThunk.fulfilled, (state, action) => {
         state.isLoading.getProjectModuleList = false;
         state.getProjectModuleList = action?.payload?.data
-          .filter((d) => d.is_active == 1)
```
Comment (somyaranjan-techneai) [inline src/redux/slices/testCases/downloadFormatFile/index.js:54]: use proper name

File: src/redux/slices/testCases/downloadFormatFile/index.js [hunk 2]
```diff
@@ -51,7 +76,7 @@ const downloadFormatSlice = createSlice({
...
+          .filter((d) => d.is_active === 1)
```
Comment (somyaranjan-techneai) [inline src/redux/slices/testCases/downloadFormatFile/index.js:79]: use proper name

File: src/redux/slices/testCases/downloadFormatFile/index.js [hunk 3]
```diff
@@ -62,41 +87,40 @@ const downloadFormatSlice = createSlice({
...
-        state.isLoading.getMouleList = false;
-        state.getModuleData = action.payload?.data.filter(
-          (d) => d.is_active == 1
-        );
-        state.getMouleList = action?.payload?.data
-          .filter((d) => d.is_active == 1)
+        state.isLoading.getModuleList = false;
+        state.getModuleData = action.payload?.data.filter((d) => d.is_active === 1);
+        state.getModuleList = action?.payload?.data
+          .filter((d) => d.is_active === 1)
```
Comment (somyaranjan-techneai) [inline src/redux/slices/testCases/downloadFormatFile/index.js:96]: use proper name

File: src/screens/TestCases/FunctionMaster/FunctionMasterComponent.js [hunk 4]
```diff
@@ -1,29 +1,25 @@
...
-  );
+  const { functionMasterList, isLoading } = useSelector((state) => state?.functionMaster);
+  const [searchValue, setSearchValue] = useState('');
+  const [filteredFunctionMasterList, setFilterFunctionMasterList] = useState([]);
 
   const [addEditFunctionModal, setAddEditFunctionModal] = useState({
-    type: "",
-    data: "",
-    open: false,
+    type: '',
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/FunctionMaster/FunctionMasterComponent.js:20]: if you use type then modal name should be according it

File: src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js [hunk 5]
```diff
@@ -1,133 +1,120 @@
...
-  const [filteredReviewCommentMasterList, setFilteredReviewCommentMasterList] =
-    useState([]);
+  const [searchValue, setSearchValue] = useState('');
+  const [filteredReviewCommentMasterList, setFilteredReviewCommentMasterList] = useState([]);
 
   const [addEditReviewCommentModal, setAddEditReviewCommentModal] = useState({
-    type: "",
-    data: "",
-    open: false,
+    type: '',
```
Comment (somyaranjan-techneai) [inline src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js:22]: if you use type then modal name should be according it

File: src/screens/TestCases/TestDraft/DownloadFormatFileModal.js [hunk 6]
```diff
@@ -144,9 +132,9 @@ function DownloadFormatFileModal({ show, close }) {
               </Row>
 
               <div className="d-flex justify-content-end gap-2 mt-3">
-                <button type="submit" className="btn btn bg-success text-white">
+                <a type="submit" href="" className="btn btn bg-success text-white">
                   Download CSV
```
Comment (AmitSolnke) [inline src/screens/TestCases/TestDraft/DownloadFormatFileModal.js:136]: 1.Not able to download format.

File: src/redux/services/testCases/testCaseReview/index.js
Comment (AmitSolnke) [inline src/redux/services/testCases/testCaseReview/index.js:1]: Multiple Api getting called, which is not necessary.

File: src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js (hunk 5 above)
Comment (AsmitaMargaje) [reply src/screens/TestCases/ReviewCommentMaster/ReviewCommentMasterComponent.js:22]: resolved

**Pull request merged**